app/src/main/java/net/skyscanner/backpack/demo/StoriesRecyclerViewAdapter.kt

```diff
@@ -60,10 +60,6 @@ class StoriesRecyclerViewAdapter internal constructor(
 
     if (values[position] is StoryItem) {
       holder.itemView.setOnClickListener(onClickListener)
-
-      if (position + 1 == itemCount || values[position + 1] is HeaderItem) {
```

Comment:
this bothered me for so long - the divider always has the wrong visibility because a) we're just hiding it here and never showing it again and b) items might not always be rebound (although that's less of an issue for a static list like this)